tests/Operation/ExplainFunctionalTest.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+namespace MongoDB\Tests\Operation;
+
+use MongoDB\Operation\Count;
+use MongoDB\Operation\Distinct;
+use MongoDB\Operation\Explain;
+use MongoDB\Operation\FindAndModify;
+use MongoDB\Operation\InsertMany;
+
+class ExplainFunctionalTest extends FunctionalTestCase
+{
+    public function setUp()
+    {
+        parent::setUp();
+        if (version_compare($this->getServerVersion(), '3.0.0', '<')) {
+            $this->markTestSkipped('Explain is not supported');
+        }
+    }
+
+    public function testCountAllPlansExecution()
+    {
+        $insertMany = new InsertMany($this->getDatabaseName(), $this->getCollectionName(), [
+            ['x' => 0],
+            ['x' => 1],
+            ['x' => 2],
+            ['y' => 3]
+        ]);
+        $insertMany->execute($this->getPrimaryServer());
+
+        $operation = new Count($this->getDatabaseName(), $this->getCollectionName(), ['x' => ['$gte' => 1]], []);
+        $explainOperation = new Explain($this->getDatabaseName(), $operation, ['verbosity' => Explain::VERBOSITY_ALL_PLANS, 'typeMap' => ['root' => 'array']]);
+        $result = $explainOperation->execute($this->getPrimaryServer());
+
+        $this->assertTrue(array_key_exists('queryPlanner', $result));
+        $this->assertTrue(array_key_exists('executionStats', $result));
+        $this->assertTrue(array_key_exists('allPlansExecution', $result['executionStats']));
+    }
+
+    public function testCountDefaultVerbosity()
+    {
+        $insertMany = new InsertMany($this->getDatabaseName(), $this->getCollectionName(), [
+            ['x' => 0],
+            ['x' => 1],
+            ['x' => 2],
+            ['y' => 3]
+        ]);
+        $insertMany->execute($this->getPrimaryServer());
+
+        $operation = new Count($this->getDatabaseName(), $this->getCollectionName(), ['x' => ['$gte' => 1]], []);
+        $explainOperation = new Explain($this->getDatabaseName(), $operation, ['typeMap' => ['root' => 'array']]);
+        $result = $explainOperation->execute($this->getPrimaryServer());
+
+        $this->assertTrue(array_key_exists('queryPlanner', $result));
+        $this->assertTrue(array_key_exists('executionStats', $result));
+        $this->assertTrue(array_key_exists('allPlansExecution', $result['executionStats']));
```

Comment:
According to the server docs, the default `verbosity` is `allPlansExecution`. I believe that makes this test very similar to `testCountAllPlansExecution()`.

In the interest of reducing code duplication throughout ExplainFunctionalTest, perhaps you can use a data provider here. I think there are a few variables:

 * `verbosity` string (if `null` is provided, your test can interpret that as an absent option)
 * Whether an `executionStats` key is expected in the result
 * Whether an `allPlansExecution` key is expected in `executionStats`

Assuming the verbosity behavior holds true for all commands, you could then write a single test method per command. To further cut down on duplication, you can also create a private `assertExplainResult()` helper that takes booleans for whether `executionStats` and `allPlansExecution` are expected.